lib/cfg-lexer.c

```diff
@@ -946,7 +946,13 @@ cfg_lexer_lex(CfgLexer *self, YYSTYPE *yylval, YYLTYPE *yylloc)
           msg_warning("WARNING: Configuration file has no version number, assuming syslog-ng 2.1 format. Please add @version: maj.min to the beginning of the file to indicate this explicitly");
           cfg_set_version(configuration, 0x0201);
         }
-      cfg_load_candidate_modules(configuration);
+
+      /* Negative would mean that this "GlobalConfig" is a partial, local config... */
+      if (self->user_version > 0) 
+        {
+          cfg_load_candidate_modules(configuration);
+        }
+
       self->non_pragma_seen = TRUE;
```

Comment:
See the description of the pull request, we are using the `GlobalConfig` architecture to load a _partial_ or _local_ config file. (I don't know, why, but we load filters from a config file in syslog-ng config file format, which contains only filter objects.)